### PR TITLE
Implement StackFrame.getDescriptor(), getMethodType()

### LINKED ISSUE #522: Add StackWalker.StackFrame.getMethodType() / getDescriptor()
StackWalker.StackFrame.getMethodType() / getDescriptor() need to be implemented for Java 18.3 / 10

### PR DESCRIPTION
**Pull request by pdbain-ibm**
Grab the signature from the ROM method and put it in the Frame object.
Use the signature to generate descriptor and MethodType.

closes #522

Signed-off-by: Peter Bain <peter_bain@ca.ibm.com>

### CODE REVIEW

**Comment by pdbain-ibm** [comment]
Tested manually with a personal build:
import java.util.HashMap;

public class StackWalk {

	public static void main(String[] args) {
		System.out.println("main");
		new StackWalk().f1(3);
	}
	
	void f1(int i) {
		if (i > 0) f1(i-1); else f2(new Integer(42));
	}

	private void f2(Integer in) {
		f3(in, new HashMap<String, Integer>());
	}

	private int f3(Integer in, HashMap<String, Integer> hashMap) {
		printStack("Hello");
		return 42;
	}

	public void printStack(String s) {
		StackWalker sw = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
		sw.forEach(f-> {System.out.println(f.getMethodName()+" getDescriptor="+f.getDescriptor()
		+" getMethodType="+f.getMethodType().toString());});
	}
}
`

result:
main
printStack getDescriptor=(Ljava/lang/String;)V getMethodType=(String)void
f3 getDescriptor=(Ljava/lang/Integer;Ljava/util/HashMap;)I getMethodType=(Integer,HashMap)int
f2 getDescriptor=(Ljava/lang/Integer;)V getMethodType=(Integer)void
f1 getDescriptor=(I)V getMethodType=(int)void
f1 getDescriptor=(I)V getMethodType=(int)void
f1 getDescriptor=(I)V getMethodType=(int)void
f1 getDescriptor=(I)V getMethodType=(int)void
main getDescriptor=([Ljava/lang/String;)V getMethodType=(String[])void

**Comment by pdbain-ibm** [comment]
Tested with an OpenJDK build, same results.

**Comment by pshipton** [comment]
We should have some basic testing for the new APIs added to the test suite, although we can't run it yet.

File: runtime/oti/vmconstantpool.xml [hunk 1]
```diff
@@ -176,6 +176,7 @@
 	<fieldref class="java/lang/StackWalker$StackFrameImpl" name="classLoaderName" signature="Ljava/lang/String;" jcl="se9_before_b165,se9"/>
 	<fieldref class="java/lang/StackWalker$StackFrameImpl" name="lineNumber" signature="I" jcl="se9_before_b165,se9"/>
 	<fieldref class="java/lang/StackWalker$StackFrameImpl" name="methodName" signature="Ljava/lang/String;" jcl="se9_before_b165,se9"/>
+	<fieldref class="java/lang/StackWalker$StackFrameImpl" name="methodSignature" signature="Ljava/lang/String;" jcl="se9_before_b165,se9"/>
```
Comment (pshipton) [inline runtime/oti/vmconstantpool.xml]: technically se9_before_b165 isn't required here, although it doesn't hurt and se9_before_b165 is obsolete anyway

**Comment by pdbain-ibm** [comment]
Peter S.: I concur. I talked to @llxia earlier today about adding tests for Java 10 but I understand there is no repo (yet) for same.

Lan and @smlambert: in re Peter Shipton's comment above, where should I put the test code for Java 10?

Thanks

**Comment by llxia** [comment]
Assume the test code will be in the same repo as jvm, we could add it as regular test in OpenJ9 except following to make sure the test will be compiled and run for Java 10.
- in build.xml, compile if JAVA_VERSION is SE10 (or SE100?)
- in playlist.xml, add subset <subset>SE10</subset>

@smlambert, do we need to create a folder `Java10andUp` to group these tests for Java 10?

**Comment by smlambert** [comment]
@llxia yes, if there are tests that are specific to Java10, then a Java10andUp is appropriate.

**Comment by pdbain-ibm** [comment]
Per Shelley's comment, lets create a bare Java10AndUp so I can check in my tests, even if we can't build & run the tests automatically (yet).  @llxia would you do the needful?
Thanks

**Comment by llxia** [comment]
@pdbain-ibm , I could check in the empty Java10andUp folder, but you still need to add src, build.xml and playlist.xml for your tests as described above. I think it would be easier if you add it all together. If you have any question, just let me know and I will be happy to help. :)

**Comment by pdbain-ibm** [comment]
Per discussion with @DanHeidinga , we suggest committing this prior to delivering the test changes in progress (see https://github.com/eclipse/openj9/pull/779).

**Comment by pdbain-ibm** [comment]
Reviewers: are there any further comments on this pull request?

**Comment by pshipton** [comment]
What's happening with the tests?

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 2]
```diff
@@ -306,6 +309,23 @@ public void forEach(Consumer<? super StackFrame> action) {
...
+		 * @return MethodType containing the parameter and return types for the associated method.
+		 */
+		default MethodType getMethodType() {
+			throw new UnsupportedOperationException();
+		}
+
+		/**
+		 * @return method descriptor string representing the type of this frame's method.
+		 */
+		default String getDescriptor() {
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: The javadoc needs to be improved here as well

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 3]
```diff
@@ -376,6 +397,23 @@ public StackTraceElement toStackTraceElement() {
 					lineNumber);
 		}
 
+		/*[IF Java10]*/
+		@Override
+		public MethodType getMethodType() {
+			if (null == declaringClass) {
+				/*[MSG "K0639","Stack walker not configured with RETAIN_CLASS_REFERENCE"]*/
+				throw new UnsupportedOperationException(com.ibm.oti.util.Msg.getString("K0639")); //$NON-NLS-1$
+			} else {
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: The `else` clause isn't required here as the `if` unconditionally throws.

File: runtime/jcl/common/java_lang_StackWalker.cpp [hunk 4]
```diff
@@ -218,6 +218,12 @@ Java_java_lang_StackWalker_getImpl(JNIEnv *env, jobject clazz, jlong walkStateP)
 			}
 			J9VMJAVALANGSTACKWALKERSTACKFRAMEIMPL_SET_METHODNAME(vmThread, PEEK_OBJECT_IN_SPECIAL_FRAME(vmThread, 0), nameString);
 
+			nameString = utfToStringObject(env, J9ROMMETHOD_GET_SIGNATURE(romClass, romMethod), J9_STR_INTERN);
```
Comment (DanHeidinga) [inline runtime/jcl/common/java_lang_StackWalker.cpp]: Why does this assign a signature to `nameString`?  A new local would be clearer

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 5]
```diff
@@ -306,6 +309,23 @@ public void forEach(Consumer<? super StackFrame> action) {
 		 * @return StackTraceElement
 		 */
 		StackTraceElement toStackTraceElement();
+
+		/*[IF Java10]*/
+		/**
+		 * @return MethodType containing the parameter and return types for the associated method.
+		 */
+		default MethodType getMethodType() {
```
[the remark this answers is not in the document]
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Thanks.  Done.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 2 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Done.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 3 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: True.  Fixed.

File: runtime/jcl/common/java_lang_StackWalker.cpp (hunk 4 above)
Comment (pdbain-ibm) [reply runtime/jcl/common/java_lang_StackWalker.cpp]: Done.
I use "nameString" as a general purpose buffer for several strings (most of them are names of something).  Ideally I should change the variable name to something more generic like "stringObject", but I'll skip that to keep the present changeset small.

**Comment by pdbain-ibm** [comment]
Dan - do you have any other changes?

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 6]
```diff
@@ -306,6 +309,29 @@ public void forEach(Consumer<? super StackFrame> action) {
 		 * @return StackTraceElement
 		 */
 		StackTraceElement toStackTraceElement();
+
+		/*[IF Java10]*/
+		/**
+		 * This method must be implemented in the implementing class.
+		 * @since 10
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Typically this goes after the @return

**Comment by pshipton** [comment]
What is the status of the testing?

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 6 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Updated.

**Comment by pdbain-ibm** [comment]
In re testing: I tests the original changeset manually, but nothing since then.
If there are no functional changes, I'll test again manually using the test material from Java10andUp.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 7]
```diff
@@ -306,6 +309,29 @@ public void forEach(Consumer<? super StackFrame> action) {
 		 * @return StackTraceElement
 		 */
 		StackTraceElement toStackTraceElement();
+
+		/*[IF Java10]*/
+		/**
+		 * This method must be implemented in the implementing class.
+		 * @throws UnsupportedOperationException if this method is not overridden
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: This should also mention that it will throw UOE unless the Stackwalker Options inclue RETAIN_CLASS_REFERENCE.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 8]
```diff
@@ -306,6 +309,29 @@ public void forEach(Consumer<? super StackFrame> action) {
 		 * @return StackTraceElement
 		 */
 		StackTraceElement toStackTraceElement();
+
+		/*[IF Java10]*/
+		/**
+		 * This method must be implemented in the implementing class.
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: This is kind of throw away text unless it clarifies the expected implementation / behaviour of the method.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 9]
```diff
@@ -306,6 +309,29 @@ public void forEach(Consumer<? super StackFrame> action) {
...
+		 * @throws UnsupportedOperationException if this method is not overridden
+		 * @return MethodType containing the parameter and return types for the associated method.
+		 * @since 10
+		 */
+		default MethodType getMethodType() {
+			throw new UnsupportedOperationException();
+		}
+
+		/**
+		 * This method must be implemented in the implementing class.
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: This is kind of throw away text unless it clarifies the expected implementation / behaviour of the method.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 10]
```diff
@@ -376,6 +403,33 @@ public StackTraceElement toStackTraceElement() {
 					lineNumber);
 		}
 
+		/*[IF Java10]*/
+		/**
+		 * Answer with a MethodType object for the method associated with this frame.
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: This reads awkwardly. Maybe "Return a MethodType describing the stackframe's method"?

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 11]
```diff
@@ -376,6 +403,33 @@ public StackTraceElement toStackTraceElement() {
...
+		public MethodType getMethodType() {
+			if (null == declaringClass) {
+				/*[MSG "K0639","Stack walker not configured with RETAIN_CLASS_REFERENCE"]*/
+				throw new UnsupportedOperationException(com.ibm.oti.util.Msg.getString("K0639")); //$NON-NLS-1$
+			}
+			return MethodType.fromMethodDescriptorString(methodSignature, declaringClass.getClassLoader());
+		}
+
+		/**
+		 * Answer with a string describing the signature of the method associated with this frame.
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Similarly awkward.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 8 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Fair comment.  Removed.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 7 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Done.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 9 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Removed.

**Comment by pshipton** [comment]
The original plan to was write some automated testing that is ready to go once we get a Java 10 extensions repo,

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 11 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Similarly fixed.  Thanks!

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 10 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Changed.  Let me know if you want further changes.

**Comment by pdbain-ibm** [comment]
The automated tests are in the Java10andUp repo. I'll test it with a recent Java 10 build (maybe from @JasonFengJ9 ) patched with my changes.

**Comment by pshipton** [comment]
The code fails to compile, see the travis-ci results

```
../common/java_lang_StackWalker.cpp:221:15: error:   crosses initialization of ‘J9Object* signatureString’
    j9object_t signatureString = utfToStringObject(env, J9ROMMETHOD_GET_SIGNATURE(romClass, romMethod), J9_STR_INTERN);
```

**Comment by pdbain-ibm** [comment]
Right.  That's probably because of the gotos. I'll fix this properly: rename nameString to a more generic name and use it throughout as mentioned above.

**Comment by pdbain-ibm** [comment]
Updated the test code and ran it on a personal build based on the Java  10 OpenJ9 builds:
java   \
	-cp "testng.jar:/team/pdbain/lib/testjars/jcommander.jar:bin/Java10andUp:TestUtilities" \
	org.testng.TestNG -d "results" "Java10andUp/testng.xml" -testnames StackWalkerTestJava10 \
	-groups level.extended \
	-excludegroups d.*.linux_x86-64,d.*.arch.x86,d.*.os.linux,d.*.bits.64,d.*.generic-all

PASSED: testRetainOption
PASSED: testSanity

===============================================
    StackWalkerTestJava10
    Tests run: 2, Failures: 0, Skips: 0
===============================================

**Comment by pdbain-ibm** [comment]
@llxia @smlambert FYI: contains test changes.

**Comment by llxia** [comment]
Test change looks good to me.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java [hunk 12]
```diff
@@ -376,6 +402,33 @@ public StackTraceElement toStackTraceElement() {
...
+		 * @return MethodType object
+		 * @since 10
+		 */
+		@Override
+		public MethodType getMethodType() {
+			if (null == declaringClass) {
+				/*[MSG "K0639","Stack walker not configured with RETAIN_CLASS_REFERENCE"]*/
+				throw new UnsupportedOperationException(com.ibm.oti.util.Msg.getString("K0639")); //$NON-NLS-1$
+			}
+			return MethodType.fromMethodDescriptorString(methodSignature, declaringClass.getClassLoader());
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/StackWalker.java]: Should `getClassLoader()` be wrapped in a `doPrivileged()` block?  Or use the package private `getClassLoaderImpl()` to avoid the security check here.

File: test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java [hunk 13]
```diff
@@ -22,45 +22,93 @@
...
+		StackFrame callerFrame = callerFrameOption.get();
+		assertEquals(callerFrame.getDescriptor(), "(Ljava/lang/StackWalker;)Ljava/lang/String;", 
+				"wrong descriptor when RETAIN_CLASS_REFERENCE not configured");
+		boolean thrown = false;
+		try {
+			callerFrame.getMethodType();
+		} catch (UnsupportedOperationException e) {
+			thrown = true;
+		}
+		assertTrue(thrown, "UnsupportedOperationException not thrown");
```
Comment (DanHeidinga) [inline test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java]: the pattern where a test is expected to throw an exception can be written more cleanly as:
```
try {
   some code..
  fail("Expected UnsupportedOperationException");
} catch(UnsupportedOperationException e) {
  /* Success */
}
```

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 12 above)
Comment (pshipton) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: > Should getClassLoader() be wrapped in a doPrivileged() block?
Not necessary as getClassLoader() checks the caller loader before doing any security check.

> Or use the package private getClassLoaderImpl() to avoid the security check here.
getClassLoaderImpl() doesn't return the same result as getClassLoader() we should add a new helper that works the same and avoid the stack walk.

File: test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java (hunk 13 above)
Comment (pdbain-ibm) [reply test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java]: Good plan.  Done.

File: jcl/src/java.base/share/classes/java/lang/StackWalker.java (hunk 12 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/StackWalker.java]: I added Class.internalGetClassLoader() which is basically getClassLoader() w/o security checks.
Please review and I will amend the changes, then re-test.

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 14]
```diff
@@ -571,6 +571,23 @@ public ClassLoader getClassLoader() {
...
+ * Returns null if the class was loaded by the bootstrap (system) class loader.
+ * This skips security checks.
+ *
+ * @return		the receiver's class loader or null
+ *
+ * @see			java.lang.ClassLoader
+ */
+ClassLoader internalGetClassLoader() {
+	ClassLoader result = classLoader;
+	if ((null != classLoader) && (classLoader == ClassLoader.bootstrapClassLoader))	{
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/Class.java]: The null check here is unnecessary.  Checking for `(classLoader == ClassLoader.bootstrapClassLoader)` is sufficient.

File: test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java [hunk 15]
```diff
@@ -22,45 +22,93 @@
...
 
 import org.testng.annotations.Test;
 import org.testng.log4testng.Logger;
 
-@Test(groups = { "level.extended" })
+/**
+ * This test StackWalker features added in Java 10 and is supplementary to the Java 9 tests.
+ *
+ */
 @SuppressWarnings("nls")
```
Comment (DanHeidinga) [inline test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java]: Don't use `@SuppressWarnings`.  Can you add the NON-NLS message to the descriptors instead?

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 16]
```diff
@@ -571,6 +571,23 @@ public ClassLoader getClassLoader() {
 	return classLoader;
 }
 
+/**
+ * Returns the classloader used to load the receiver's class.
+ * Returns null if the class was loaded by the bootstrap (system) class loader.
+ * This skips security checks.
+ *
+ * @return		the receiver's class loader or null
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/Class.java]: Can you remove the extra spacing here and below?  It's unnecessary

File: jcl/src/java.base/share/classes/java/lang/Class.java (hunk 14 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/Class.java]: Okay.

**Comment by pdbain-ibm** [comment]
Do the reviewers have any other comments?

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 17]
```diff
@@ -571,6 +571,17 @@ public ClassLoader getClassLoader() {
 	return classLoader;
 }
 
+/**
+ * Returns the classloader used to load the receiver's class.
+ * Returns null if the class was loaded by the bootstrap (system) class loader.
+ * This skips security checks.
+ * @return		the receiver's class loader or null
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/Class.java]: These big indents between the `@return` tag and the text are unnecessary.  I know they pervade the existing code but they're not something we should be propagating.

File: jcl/src/java.base/share/classes/java/lang/Class.java [hunk 18]
```diff
@@ -571,6 +571,17 @@ public ClassLoader getClassLoader() {
...
+ * @see			java.lang.ClassLoader
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/Class.java]: Same here

File: test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java [hunk 19]
```diff
@@ -22,45 +22,92 @@
...
-		.walk(s -> s.collect(Collectors.toList()));
-		String expectedDescriptors[];
-		for (StackFrame f: frameList) {
-			System.out.println(f.getDescriptor());
+		final ClassLoader myClassLoader = this.getClass().getClassLoader();
+		List<StackFrame> frameList = 
+				StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE)
+				.walk(s -> s.collect(Collectors.toList()));
+		String expectedDescriptors[] = {
+				"(Ljava/lang/String;)V",
```
Comment (DanHeidinga) [inline test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java]: Shouldn't these strings have `//NON-NLS` comments?

File: test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java [hunk 20]
```diff
@@ -22,45 +22,92 @@
...
+				"()V"
+		};
+		Iterator<StackFrame> frameIterator = frameList.iterator();
+		for (String s: expectedDescriptors) {
+			assertTrue(frameIterator.hasNext(), "too few frames");
+			StackFrame f = frameIterator.next();
+			final String actualDescriptor = f.getDescriptor();
+			assertEquals(actualDescriptor, s, "wrong descriptor");
+			assertEquals(MethodType.fromMethodDescriptorString(s,myClassLoader), f.getMethodType(), 
+					"wrong descriptor");
```
Comment (DanHeidinga) [inline test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java]: Same here and throughout?

File: jcl/src/java.base/share/classes/java/lang/Class.java (hunk 17 above)
Comment (pdbain-ibm) [reply jcl/src/java.base/share/classes/java/lang/Class.java]: Fixed.

File: test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java (hunk 20 above)
Comment (pdbain-ibm) [reply test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java]: Fixed.

File: test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java (hunk 19 above)
Comment (pdbain-ibm) [reply test/Java10andUp/src/org/openj9/test/stackWalker/StackWalkerTestJava10.java]: The project settings were not configured to warn about non-externalized strings.
Settings updated, comments added.

File: test/Java10andUp/.settings/org.eclipse.jdt.core.prefs [hunk 21]
```diff
@@ -1,11 +1,100 @@
 eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.annotation.inheritNullAnnotations=disabled
```
Comment (DanHeidinga) [inline test/Java10andUp/.settings/org.eclipse.jdt.core.prefs]: Can you remove this change from this PR?  Updating the eclipse jdt settings should be done in a separate pull request and should be done for all the JavaXandUp directories at the same time.

Comment (pdbain-ibm) [reply test/Java10andUp/.settings/org.eclipse.jdt.core.prefs]: Done.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux
